pulser-core/pulser/sampler/samples.py

```diff
@@ -100,9 +109,11 @@ def extend_duration(self, new_duration: int) -> ChannelSamples:
         Returns:
             The extend channel samples.
         """
-        extension = new_duration - len(self.amp)
-        if new_duration < self.duration:
+        extension = new_duration - self.duration
+        if extension < 0:
             raise ValueError("Can't extend samples to a lower duration.")
+        if not extension:
+            return self
```

Comment:
Should it be a copy of self, to be consistent with returning a new object?

Reply:
You're right, I agree this is abusive... I just added this because `extend_duration` was being called blindly in `SequenceSamples.to_nested_dict()`. I think I'll remove this case and instead call `extend_duration()` in `SequenceSamples.to_nested_dict()` only when needed. Sounds good?

Reply:
Sure!